pipeline/tests/api_test.ml

```diff
@@ -0,0 +1,139 @@
+let example_conf =
+  [
+    { Config.repo = "myowner/myrepo"; token = "p455word!" };
+    { Config.repo = "other/fst"; token = "token-should-be-unique" };
+    { Config.repo = "other/snd"; token = "token-should-be-unique" };
+  ]
+
+(* TODO: Cohttp should not be required for testing auth *)
```

Comment:
Why should `Cohttp` not be used? Because we want to send dummy requests instead? :)

Reply:
I guess so. I think we could refactor the validation code to use the token string directly instead of the request to make the testing easier.

Reply:
It's just a bit weird that we have to simulate a cohttp request for all of the tests. It makes sense for `invalid_bearer` (because it is testing an invalid header), but for example `find_repository` doesn't really need it, we only want to test the token validation here and not the extraction of the token from the headers. (it makes it a bit easier to understand a problem when just a few tests can independently fail)